python/cuspatial/cuspatial/core/binpreds/contains.py

```diff
@@ -12,7 +15,69 @@
 from cuspatial.utils.column_utils import normalize_point_columns
 
 
-def contains_properly(
+def contains_properly_quadtree(
+    test_points,
+    polygons,
+):
+    """Compute from a series of points and a series of polygons which points
+    are properly contained within the corresponding polygon. Polygon A contains
+    Point B properly if B intersects the interior of A but not the boundary (or
+    exterior).
+
+    Note that polygons must be closed: the first and last vertex of each
+    polygon must be the same.
+
+    Parameters
+    ----------
+    test_points
+        GeoSeries of points to test for containment
+    polygons
+        GeoSeries of polygons to test for containment
+
+    Returns
+    -------
+    result : cudf.Series
+        A Series of boolean values indicating whether each point falls
+        within its corresponding polygon.
+    """
+
+    scale = -1
+    max_depth = 15
+    min_size = ceil(sqrt(len(test_points)))
+    x_max = polygons.polygons.x.max()
+    x_min = polygons.polygons.x.min()
+    y_max = polygons.polygons.y.max()
+    y_min = polygons.polygons.y.min()
+    point_indices, quadtree = cuspatial.quadtree_on_points(
+        test_points,
+        x_min,
+        x_max,
+        y_min,
+        y_max,
+        scale,
+        max_depth,
+        min_size,
+    )
+    poly_bboxes = cuspatial.polygon_bounding_boxes(polygons)
+    intersections = cuspatial.join_quadtree_and_bounding_boxes(
+        quadtree, poly_bboxes, x_min, x_max, y_min, y_max, scale, max_depth
+    )
+    polygons_and_points = cuspatial.quadtree_point_in_polygon(
+        intersections,
+        quadtree,
+        point_indices,
+        test_points,
+        polygons,
+    )
+    polygons_and_points["point_index"] = point_indices.iloc[
+        polygons_and_points["point_index"]
+    ].reset_index(drop=True)
+    polygons_and_points["part_index"] = polygons_and_points["polygon_index"]
+    polygons_and_points.drop("polygon_index", axis=1, inplace=True)
+    return polygons_and_points
+
+
+def contains_properly_pairwise(
```

Comment:
This wasn't covered in geoseries refactor... Do you want to convert it to geoseries input?

Reply:
Sure!